python/lsst/dax/obscore/obscore_exporter.py

```diff
@@ -327,32 +332,38 @@ def _make_record_batches(self, batch_size: int = 10_000) -> Iterator[RecordBatch
             collections = ...
 
         registry = self.butler.registry
-        for dataset_type_name in self.config.dataset_types:
+        assert isinstance(registry, SqlRegistry), "Registry must be SqlRegistry"
+        backend = SqlQueryBackend(registry._db, registry._managers)
+
+        with backend.context() as context:
```

Comment:
You don't necessarily need to enter the context here; that will maintain a connection (and a transaction, too for PostgreSQL) to let you use temporary tables as part of the query, but if that's not desirable you should just do
```
context = backend.context()
```
here.